core/src/main/java/bisq/core/dao/burningman/accounting/node/full/AccountingFullNode.java

```diff
@@ -0,0 +1,323 @@
+import bisq.core.dao.burningman.accounting.exceptions.BlockHashNotConnectingException;
+import bisq.core.dao.burningman.accounting.exceptions.BlockHeightNotConnectingException;
+import bisq.core.dao.burningman.accounting.node.AccountingNode;
+import bisq.core.dao.burningman.accounting.node.full.network.AccountingFullNodeNetworkService;
+import bisq.core.dao.node.full.RpcException;
+import bisq.core.dao.node.full.RpcService;
+import bisq.core.dao.node.full.rpc.NotificationHandlerException;
+import bisq.core.dao.node.full.rpc.dto.RawDtoBlock;
+import bisq.core.dao.state.DaoStateService;
+
+import bisq.network.p2p.P2PService;
+
+import bisq.common.UserThread;
+import bisq.common.handlers.ResultHandler;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.net.ConnectException;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class AccountingFullNode extends AccountingNode {
+    private final RpcService rpcService;
+    private final AccountingFullNodeNetworkService accountingFullNodeNetworkService;
+
+    private boolean addBlockHandlerAdded;
+    private int batchedBlocks;
+    private long batchStartTime;
+    private final List<RawDtoBlock> pendingRawDtoBlocks = new ArrayList<>();
+    private int requestBlocksUpToHeadHeightCounter;
+
+    @Inject
+    public AccountingFullNode(P2PService p2PService,
+                              DaoStateService daoStateService,
+                              BurningManAccountingService burningManAccountingService,
+                              AccountingBlockParser accountingBlockParser,
+                              AccountingFullNodeNetworkService accountingFullNodeNetworkService,
+                              RpcService rpcService) {
+        super(p2PService, daoStateService, burningManAccountingService, accountingBlockParser);
+
+        this.rpcService = rpcService;
+        this.accountingFullNodeNetworkService = accountingFullNodeNetworkService;
+    }
+
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // Public methods
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    @Override
+    public void start() {
+        // We do not start yes but wait until DAO block parsing is complete to not interfere with
+        // that higher priority activity.
+    }
+
+    @Override
+    public void shutDown() {
+        accountingFullNodeNetworkService.shutDown();
+    }
+
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // Protected
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    // We start after initial DAO parsing is complete
+    @Override
+    protected void onInitialDaoBlockParsingComplete() {
```

Comment:
It makes sense to start this after dao parsing is done, but I feel it's hard to get an understanding for when services are run. Now it's not a big issue with just a couple staggered like this, but as a principle it would probably be better to have a startup handler that did the scheduling for easier overview.

Reply:
Yes, that got a bit messy over time. But I don't want to do bigger refactorings there.